code/ui/ui_main.c

```diff
@@ -737,14 +759,22 @@ void _UI_Refresh( int realtime )
 		UI_BuildServerStatus(qfalse);
 		// refresh find player list
 		UI_BuildFindPlayerList(qfalse);
-	} 
-	
-	// draw cursor
-	UI_SetColor( NULL );
-	if (Menu_Count() > 0) {
-		UI_WideScreenMode(qtrue);
-		UI_DrawHandlePic(uiInfo.uiDC.cursorx * uiInfo.screenXFactorInv, uiInfo.uiDC.cursory, 48, 48, uiInfo.uiDC.Assets.cursor);
-		UI_WideScreenMode(qfalse);
+
+		// draw cursor
+		if ( (trap_Key_GetCatcher() & KEYCATCH_UI) && Menu_Count() > 0 ) {
+			UI_SetColor(NULL);
+			if ( (ui_widescreenCursorScale.integer & 2) && mvapi >= 3 && uiInfo.portraitMode && isMainMenu ) { //Scale height on the main menu in portrait mode.
+				trap_MVAPI_SetVirtualScreen((float)SCREEN_WIDTH, (float)SCREEN_HEIGHT * uiInfo.screenYFactorInv);
+				UI_DrawHandlePic(uiInfo.uiDC.cursorx, uiInfo.uiDC.cursory * uiInfo.screenYFactorInv, 48, 48, uiInfo.uiDC.Assets.cursor);
+			} else if ( (ui_widescreenCursorScale.integer & 1) || uiInfo.portraitMode ) { // Ingame portrait mode is a special case, as we draw the menu only on a portion of the screen
```

Comment:
This is why I preferred to have ui_sensitivity - to avoid externalizing portraitMode outside of widescreen implementation context (UI_WidescreenMode()) here.
Also I'm not sure but it looks to me like ui mouse sensitivity does not depend on /sensitivity cvar. This means that people with 10000 dpi mouse could benefit from ui_sensitivity cvar greatly